### ISSUE TITLE
Enhancement: [restrict-plus-operands] no error when using () for grouping numbers and strings

### ISSUE THREAD

**Issue by kkmuffme** [typescript-eslint/typescript-eslint#7034]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/restrict-plus-operands/

### Description

When using () to group numeric operations and then concat with strings, the developer is aware that we are concat strings with numbers (otherwise wouldn't have used the () in the first place)
Therefore, this rule shouldn't report an error, mostly to avoid 1000s of redundant .toString() calls in the code base

### Fail

```typescript
no change
```

### Pass

```typescript
const pass = 'hello' + ( 15 + 3 ) + 'world';
```
